AdaptiveRoads/UI/QuayRoads/QuayRoadsPanelField.cs

```diff
@@ -0,0 +1,111 @@
+                            assetValue_ = (assetValue_ as IConvertible).ToInt32(CultureInfo.InvariantCulture) | flag.ToInt32(CultureInfo.InvariantCulture);
+                        } else {
+                            assetValue_ = (assetValue_ as IConvertible).ToInt64() | flag.ToInt64();
+                        }
+                    } else {
+                        if (Enum.GetUnderlyingType(fieldInfo_.FieldType) == typeof(Int32)) {
+                            assetValue_ = (assetValue_ as IConvertible).ToInt32(CultureInfo.InvariantCulture) & ~flag.ToInt32(CultureInfo.InvariantCulture);
+                        } else {
+                            assetValue_ = (assetValue_ as IConvertible).ToInt64() &~flag.ToInt64();
+                        }
+                    }
+                };
+            } else {
+                // TODO: right now, this assumes everything that is not a flags enum is a float
+                // TODO: find a better way to create a text field - maybe something similar to UICheckBoxExt from KianCommons?
+                var propertyHelper = new UIHelper(parent);
+                var propertyTextField = propertyHelper.AddTextfield("wawa", (assetValue_ as float?).ToString(), (_) => { }, (_) => { }) as UITextField;
+                var labelObject = parent.Find<UILabel>("Label");
+                labelObject.parent.RemoveUIComponent(labelObject);
+                Destroy(labelObject.gameObject);
+                (propertyTextField.parent as UIPanel).autoFitChildrenHorizontally = true;
+                (propertyTextField.parent as UIPanel).autoFitChildrenVertically = true;
+                propertyTextField.numericalOnly = true;
+                propertyTextField.allowFloats = true;
+                propertyTextField.allowNegative = true;
+                propertyTextField.eventTextSubmitted += (_, value) => {
+                    float newValue = (float)lenientStringToDouble_(value, (double)(float)assetValue_);
+                    propertyTextField.text = newValue.ToString();
+                    if (newValue != (float)assetValue_) {
+                        assetValue_ = newValue;
+                    }
+                };
+            }
+        }
+        private ProfileSection profileSection_ {
+            get => netInfo_.GetMetaData().quayRoadsProfile[sectionIndex_];
+            set => netInfo_.GetMetaData().quayRoadsProfile[sectionIndex_] = value;
+        }
+        private object assetValue_ {
+            get => fieldInfo_.GetValue(profileSection_);
+            set {
+                object tmp = profileSection_;
+                fieldInfo_.SetValue(tmp, value);
+                profileSection_ = (ProfileSection) tmp;
+                Log.Debug(fieldInfo_.Name + " set to " + assetValue_ + " should new value: " + value);
+                parentPanel_.OnObjectModified();
+            }
+        }
+
+        private static double lenientStringToDouble_(string s, double fallback) {
```

Comment:
[low priority ] functions (Even private ones) should be CammelCase.

Reply:
resolved in next commit.